stl/inc/regex

```diff
@@ -1820,12 +1820,14 @@ public:
     }
 
     basic_regex(_In_reads_(_Count) const _Elem* _Ptr, size_t _Count, flag_type _Flags = regex_constants::ECMAScript) {
-        if (_Ptr) {
-            _Reset(_Ptr, _Ptr + _Count, _Flags);
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(_Ptr != nullptr || _Count == 0, "constructing basic_regex from an invalid range");
+#else // ^^^ _ITERATOR_DEBUG_LEVEL != 0 / _ITERATOR_DEBUG_LEVEL == 0 vvv
+        if (_Ptr == nullptr && _Count != 0) {
             return;
         }

```

Comment:
This doesn't look right - it's doing checking in IDL=0, *and* tolerating precondition violations.